steam_wrapper/steam_wrapper.py

```diff
@@ -215,8 +216,10 @@ def migrate_config():
     migrator = Migrator(os.path.expandvars("$XDG_CONFIG_HOME"),
                         os.path.join(FLATPAK_STATE_DIR, DEFAULT_CONFIG_DIR),
                         two_steps=True)
-    migrator.apply()
-    os.environ["XDG_CONFIG_HOME"] = os.path.expandvars(f"$HOME/{DEFAULT_CONFIG_DIR}")
+    should_restart = migrator.apply()
+    if not should_restart:
```

Comment:
I'm not sure why we need this check? AFAIK `flatpak-spawn` resets environment anyway.

Reply:
Because we want to avoid infinite sandbox recursion. We need to stop re-launching the wrapper script in new sandbox when migration is done.

Reply:
But setting `XDG_CONFIG_HOME` doesn't affect restart.

Reply:
Probably my comment should've been attached to the next line. Actual question if: why set `XDG_CONFIG_HOME` only if we don't need restart?

Reply:
Tested, not needed.